[MOV] web, calendar: move common scss into web

Brieuc-brd opened the pull request:
### Move common scss into web
Prior to this PR, some of the main styles related to event elements in calendars were set in the `calendar` module.
This created layout issues when we installed independent modules without installing `calendar` (eg. `planning` or `hr_payroll`).

To fix that, this PR moves the necessary styles into `web` and manages duplicated styles in order to remove them.

### Adjust hatched events for planning
Prior to this commit, the "hatched" event style in Planning wasn't displayed in Day and Week views.
This commit adjusts the css selector to fix this issue.

---
task-3617329
Part of task-3575827

---
| Before | After |
|--------|--------|
| ![Capture d’écran 2024-02-22 à 08 30 37](https://github.com/odoo/odoo/assets/80679690/a6a33398-522e-471c-ab75-884f72c214cc) | ![Capture d’écran 2024-02-22 à 08 43 32](https://github.com/odoo/odoo/assets/80679690/e6c07aac-8273-4772-a759-64c03ee125e5) |
| ![Capture d’écran 2024-02-22 à 08 31 10](https://github.com/odoo/odoo/assets/80679690/0ac54c01-a39b-45dd-afeb-5bcc29c912ae) | ![Capture d’écran 2024-02-22 à 08 44 50](https://github.com/odoo/odoo/assets/80679690/d0a69d3f-e68f-4c7f-97d1-e33c14c6f8bd) | 

| Before | After |
|--------|--------|
| ![Capture d’écran 2024-03-25 à 11 00 46](https://github.com/odoo/odoo/assets/80679690/d866eadf-5c04-451c-a98e-4af7789f6d08) | ![Capture d’écran 2024-03-25 à 11 01 06](https://github.com/odoo/odoo/assets/80679690/216ddadb-8e02-4522-a7dc-cf3f6f71069e) |


---

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/154964).

Brieuc-brd commented:
Hey @stefanorigano 👋 
The related task targets `master` but I think we can solve this issue since `17.0`, could you take a look at this PR? :)

stefanorigano commented on addons/calendar/static/src/views/attendee_calendar/common/attendee_calendar_common_renderer.js:
```diff
@@ -46,6 +46,9 @@ export class AttendeeCalendarCommonRenderer extends CalendarCommonRenderer {
             } else {
                 el.classList.add(`o_attendee_status_${record.attendeeStatus}`);
+                if(record.attendeeStatus === 'tentative') {
+                    el.classList.add('o_event_hatched');
```
😉

```suggestion
                    el.classList.toggle('o_event_hatched', record.attendeeStatus === 'tentative');
```

stefanorigano reviewed (commented):
Hello @Brieuc-brd , it definitely makes sense and it should be doable in `17.0` I think.
A suggestion for you in the code.

Also, you should replace one of the commit prefix to `[MOV]` ([ref.](https://www.odoo.com/documentation/master/contributing/development/git_guidelines.html#tag-and-module-name)).

Let's not forget to keep @bouvyd in the loop about this.
The PR looks quite straightforward by a code/design stand point, but this kind of changes should be deeply tested, especially in prod.

Brieuc-brd commented:
Hello @bouvyd 👋 ,
I'd like to know what you think of this PR and if you can test it?

On my side, I did a global search for all modules that use the calendar and found theses modules :
- `planning`
- `hr_appraisal`
- `calendar`
- `appointment`
- `helpdesk`
- `hr_holidays`
- `hr_payroll`
- `crm`
- `hr`
- `hr_recruitement`

In this list, only `planning` & `hr_payroll` seem to not have `calendar` as a dependency, so I tested these modules in particular. But maybe I missed something ?

Brieuc-brd commented:
Hello @bouvyd 👋 
Little reminder :)

Brieuc-brd commented:
Hello @bouvyd 👋
Little reminder :)

bouvyd commented:
@Brieuc-brd everything checks out on my end, ok for me!

Brieuc-brd commented:
Hello @odoo/rd-subscription-sign-calendar @odoo/rd-framework-js 

I'm not sure who I should ping on this, but this PR is ready for review, could one of you take a look at it? :)

pparidans reviewed (approved):
LGTM and make the calendar view styling more "predictable" :+1:

pparidans commented:
@robodoo rebase-ff r+

robodoo commented:
Merge method set to rebase and fast-forward.

robodoo commented:
@Brieuc-brd @pparidans unable to stage: merge conflict

pparidans commented:
@Brieuc-brd may I ask you to rebase your branch, please ?

Brieuc-brd commented:
Hey @pparidans 👋 
I rebased the branch and fixed the conflicts.
I also took the opportunity to fix an issue related to the "hatched" style of events in Planning.

Could you take a look at it?

pparidans reviewed (approved):
Thx @Brieuc-brd ! LGTM

@robodoo rebase-ff r+
